composer/datasets/in_context_learning_evaluation.py

```diff
@@ -296,6 +296,14 @@ def __init__(
         self.max_seq_len = max_seq_len
         self.pad_tok_id = pad_tok_id
         fewshot_rng = random.Random(fewshot_random_seed)
+
+        # Test for whether a prefix space is needed before the continuation
+        # sentencepiece tokenization should not have a prefix space, but gpt2 style BPE should
+        self.prefix_space = True
+        test = self.tokenizer(' a', add_special_tokens=False)['input_ids']
+        if len(test) > 1:
+            self.prefix_space = False
```

Comment:
should this be a `tokenizer_needs_prefix_space` function or something? I could see us needing this more, and it already shows up twice. Not a hard requirement

Reply:
yeah, very reasonable, will change